src/libostree/ostree-repo-pull.c

```diff
@@ -1369,6 +1537,44 @@ scan_commit_object (OtPullData         *pull_data,
   if (!ostree_repo_load_commit (pull_data->repo, checksum, &commit, &commitstate, error))
     goto out;
 
+  /* If ref is NULL it means that the commit we fetched is the commit
```

Comment:
Ah, no. Maybe it should be rephrased like "if ref is not NULL then the commit we fetched was requested through the branch, otherwise we requested a commit checksum without specifying a branch".